src/main/java/se/sundsvall/emailreader/service/EmailScheduler.java

```diff
@@ -68,7 +76,23 @@ public void scheduleEmailReader() {
                 }
             });
         }));
-        log.info("Finished scheduled email reader");
+    }
+
+    @Scheduled(initialDelayString = "${scheduled.email-age-check.initial-delay}", fixedRateString = "${scheduled.email-age-check.fixed-rate}")
+    void checkForOldEmails() {
+
+        final var list = emailRepository.findAll().stream()
```

Comment:
Maybe not a million emails, but for performance, perhaps it would be an idea to not fetch all emails?